### PR TITLE
feat(scan-type): AWS Resource ARNs based scan

### PR DESCRIPTION
**Pull request by MrCloudSec**
### Description

AddAWS Resource ARNs based scan using `--resource-arn` followed by one or more [Amazon Resource Names (ARNs)](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html) separated by space:

```
prowler aws --resource-arn arn:aws:iam::012345678910:user/test arn:aws:ec2:us-east-1:123456789012:vpc/vpc-12345678
```

This example will only scan the two resources with those ARNs.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jfagoagas — COMMENTED** [review]
Please include tests for the new functions and the parser.

File: prowler/lib/check/checks_loader.py [hunk 1]
```diff
@@ -18,10 +19,17 @@ def load_checks_to_execute(
...
 ) -> set:
     """Generate the list of checks to execute based on the cloud provider and input arguments specified"""
     checks_to_execute = set()
 
+    # Handle if there are audit resources so only their services are executed
+    if audit_info.audit_resources:
+        service_list = []
+        for resource in audit_info.audit_resources:
+            service_list.append(resource.split(":")[2])
+
```
Comment (jfagoagas) [inline prowler/lib/check/checks_loader.py]: So awesome!!!!

File: prowler/lib/cli/parser.py [hunk 2]
```diff
@@ -343,14 +350,25 @@ def __init_aws_parser__(self):
...
             nargs="+",
             default=None,
-            help="Scan only resources with specific tags (Key=Value), e.g., Environment=dev Project=prowler",
+            help="Scan only resources with specific AWS Tags (Key=Value), e.g., Environment=dev Project=prowler",
+        )
+        aws_based_scans_parser.add_argument(
+            "-r",
+            "--resource-arns",
+            nargs="+",
+            type=arn_type,
```
Comment (jfagoagas) [inline prowler/lib/cli/parser.py:368]: What is this?

Comment (MrCloudSec) [reply prowler/lib/cli/parser.py:368]: It is a custom type that calls the `arn_type` function to validate each input ARN.

Comment (jfagoagas) [reply prowler/lib/cli/parser.py:368]: awesone, even with a list?

Comment (MrCloudSec) [reply prowler/lib/cli/parser.py:368]: yeah 👍🏼

File: tests/lib/cli/parser_test.py [hunk 3]
```diff
@@ -796,23 +796,25 @@ def test_aws_parser_allowlist_long(self):
...
+
+    def test_aws_parser_resource_arn(self):
+        argument = "--resource-arn"
+        resource_arn1 = "arn:aws:iam::012345678910:user/test"
+        resource_arn2 = "arn:aws:ec2:us-east-1:123456789012:vpc/vpc-12345678"
+        command = [prowler_command, argument, resource_arn1, resource_arn2]
+        parsed = self.parser.parse(command)
+        assert len(parsed.resource_arn) == 2
+        assert resource_arn1 in parsed.resource_arn
+        assert resource_arn2 in parsed.resource_arn
```
Comment (jfagoagas) [inline tests/lib/cli/parser_test.py:817]: Can you include a test with a wrong ARN as argument? Just to validate it.

You can use the following:
```
with pytest.raises(argparse.ArgumentError) as ex:
  parsed = self.parser.parse(command)

assert ex.type == argparse.ArgumentError
assert ex.message == "Invalid ARN"
```

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: tests/lib/cli/parser_test.py (hunk 3 above)
Comment (MrCloudSec) [reply tests/lib/cli/parser_test.py:817]: Done

File: prowler/lib/cli/parser.py [hunk 4]
```diff
@@ -4,6 +4,13 @@
 
 from prowler.config.config import default_output_directory, prowler_version
 from prowler.providers.aws.aws_provider import get_aws_available_regions
+from prowler.providers.aws.lib.arn.arn import is_valid_arn
+
+
+def arn_type(arn: str):
```
Comment (jfagoagas) [inline prowler/lib/cli/parser.py]:
```suggestion
def arn_type(arn: str) -> bool:
  """ arn_type returns a string ARN if it is valid and raises an argparse.ArgumentError if not. """
```

File: prowler/providers/aws/lib/arn/arn.py [hunk 5]
```diff
@@ -43,3 +45,9 @@ def arn_parsing(arn):
                 raise RoleArnParsingEmptyResource
             else:
                 return arn_parsed
+
+
+def is_valid_arn(arn):
```
Comment (jfagoagas) [inline prowler/providers/aws/lib/arn/arn.py]:
```suggestion
def is_valid_arn(arn: str) -> bool:
  """ is_valid_arn returns True or False whether the given AWS ARN (Amazon Resource Name) is valid or not. """
```

File: prowler/providers/aws/lib/arn/arn.py [hunk 6]
```diff
@@ -43,3 +45,9 @@ def arn_parsing(arn):
...
+    # checks if argument is a valid ARN (Amazon Resource Name)
```
Comment (jfagoagas) [inline prowler/providers/aws/lib/arn/arn.py]:
```suggestion
```

**Review by jfagoagas — DISMISSED** [review]

**Review by jfagoagas — APPROVED** [review]

**Pull request merged**